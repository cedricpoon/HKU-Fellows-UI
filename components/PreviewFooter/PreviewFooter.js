import React, { Component } from 'react';
import { Footer, FooterTab, Button, Icon } from 'native-base';
import { withNavigation } from 'react-navigation';

import FilterPopup from './FilterPopup/FilterPopup';
import { mapLayoutToState } from 'hkufui/components/helper';
import styles from './Styles';

class PreviewFooter extends Component {

  constructor(props) {
    super(props);
    this.state = { showFilterPopup: false, footerLayout: { height: 0 } };

    this.filterToggle = this.filterToggle.bind(this);
  }

  filterToggle() {
    this.setState(prevState => ({
      showFilterPopup: !prevState.showFilterPopup
    }));
  }

  renderFilterPopup() {
    const { filterLayout, showFilterPopup, footerLayout } = this.state;

    if (showFilterPopup) {
      return(
        <FilterPopup
          position={filterLayout.x}
          parentHeight={footerLayout.height}
          toggle={this.filterToggle}
        />
      );
    }
    return null;
  }

  render() {
    return (
      <Footer style={styles.footer}>
        { this.renderFilterPopup() }
        <FooterTab
          onLayout={mapLayoutToState("footerLayout", this)}
        >
          <Button>
            <Icon name="menu" />
          </Button>
          <Button>
            <Icon name="refresh" />
          </Button>
          <Button>
            <Icon name="create" />
          </Button>
          <Button
            onPress={this.filterToggle}
            onLongPress={this.filterToggle}
            onLayout={mapLayoutToState("filterLayout", this)}
          >
            <Icon name="funnel"></Icon>
          </Button>
          <Button
            onPress={()=>{ this.props.navigation.navigate('SelectCourse') }}
          >
            <Icon name="albums" />
          </Button>
        </FooterTab>
      </Footer>
    );
  }
}

export default withNavigation(PreviewFooter);
