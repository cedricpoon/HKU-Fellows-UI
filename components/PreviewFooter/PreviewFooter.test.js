import React from 'react';
import { shallow } from 'enzyme';
import PreviewFooter from './PreviewFooter';

describe('Testing PreviewFooter Component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <PreviewFooter />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders in muted', () => {
    const wrapper = shallow(
      <PreviewFooter muted />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
