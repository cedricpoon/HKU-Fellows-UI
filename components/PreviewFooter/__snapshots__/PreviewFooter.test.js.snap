// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PreviewFooter Component renders as expected with requiredProps 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PreviewFooter
    onRefresh={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <FilterPopup
          onRef={[Function]}
          position={
            Object {
              "x": 0,
              "y": 0,
            }
          }
        />,
        <Styled(FooterTab)>
          <Styled(Button)
            onPress={[Function]}
          >
            <Styled(Icon)
              name="menu"
              type="MaterialIcons"
            />
          </Styled(Button)>
          <Styled(Button)
            onPress={[Function]}
          >
            <Styled(Icon)
              name="refresh"
              type="MaterialIcons"
            />
          </Styled(Button)>
          <Styled(Button)>
            <Styled(Icon)
              name="add"
              type="MaterialIcons"
            />
          </Styled(Button)>
          <Styled(Button)
            onLayout={[Function]}
            onLongPress={[Function]}
            onPress={[Function]}
          >
            <Styled(Icon)
              name="sort-descending"
              type="MaterialCommunityIcons"
            />
          </Styled(Button)>
          <Styled(Button)
            onPress={[Function]}
          >
            <Styled(Icon)
              name="view-list"
              type="MaterialIcons"
            />
          </Styled(Button)>
        </Styled(FooterTab)>,
      ],
      "onLayout": [Function],
      "style": Object {
        "zIndex": 1,
      },
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onRef": [Function],
          "parentHeight": undefined,
          "position": Object {
            "x": 0,
            "y": 0,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Button)
              onPress={[Function]}
            >
              <Styled(Icon)
                name="menu"
                type="MaterialIcons"
              />
            </Styled(Button)>,
            <Styled(Button)
              onPress={[Function]}
            >
              <Styled(Icon)
                name="refresh"
                type="MaterialIcons"
              />
            </Styled(Button)>,
            <Styled(Button)>
              <Styled(Icon)
                name="add"
                type="MaterialIcons"
              />
            </Styled(Button)>,
            <Styled(Button)
              onLayout={[Function]}
              onLongPress={[Function]}
              onPress={[Function]}
            >
              <Styled(Icon)
                name="sort-descending"
                type="MaterialCommunityIcons"
              />
            </Styled(Button)>,
            <Styled(Button)
              onPress={[Function]}
            >
              <Styled(Icon)
                name="view-list"
                type="MaterialIcons"
              />
            </Styled(Button)>,
          ],
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Icon)
                name="menu"
                type="MaterialIcons"
              />,
              "onPress": [Function],
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "menu",
                "type": "MaterialIcons",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Icon)
                name="refresh"
                type="MaterialIcons"
              />,
              "onPress": [Function],
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "refresh",
                "type": "MaterialIcons",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Icon)
                name="add"
                type="MaterialIcons"
              />,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "add",
                "type": "MaterialIcons",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Icon)
                name="sort-descending"
                type="MaterialCommunityIcons"
              />,
              "onLayout": [Function],
              "onLongPress": [Function],
              "onPress": [Function],
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "sort-descending",
                "type": "MaterialCommunityIcons",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Icon)
                name="view-list"
                type="MaterialIcons"
              />,
              "onPress": [Function],
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "view-list",
                "type": "MaterialIcons",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <FilterPopup
            onRef={[Function]}
            position={
              Object {
                "x": 0,
                "y": 0,
              }
            }
          />,
          <Styled(FooterTab)>
            <Styled(Button)
              onPress={[Function]}
            >
              <Styled(Icon)
                name="menu"
                type="MaterialIcons"
              />
            </Styled(Button)>
            <Styled(Button)
              onPress={[Function]}
            >
              <Styled(Icon)
                name="refresh"
                type="MaterialIcons"
              />
            </Styled(Button)>
            <Styled(Button)>
              <Styled(Icon)
                name="add"
                type="MaterialIcons"
              />
            </Styled(Button)>
            <Styled(Button)
              onLayout={[Function]}
              onLongPress={[Function]}
              onPress={[Function]}
            >
              <Styled(Icon)
                name="sort-descending"
                type="MaterialCommunityIcons"
              />
            </Styled(Button)>
            <Styled(Button)
              onPress={[Function]}
            >
              <Styled(Icon)
                name="view-list"
                type="MaterialIcons"
              />
            </Styled(Button)>
          </Styled(FooterTab)>,
        ],
        "onLayout": [Function],
        "style": Object {
          "zIndex": 1,
        },
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onRef": [Function],
            "parentHeight": undefined,
            "position": Object {
              "x": 0,
              "y": 0,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(Button)
                onPress={[Function]}
              >
                <Styled(Icon)
                  name="menu"
                  type="MaterialIcons"
                />
              </Styled(Button)>,
              <Styled(Button)
                onPress={[Function]}
              >
                <Styled(Icon)
                  name="refresh"
                  type="MaterialIcons"
                />
              </Styled(Button)>,
              <Styled(Button)>
                <Styled(Icon)
                  name="add"
                  type="MaterialIcons"
                />
              </Styled(Button)>,
              <Styled(Button)
                onLayout={[Function]}
                onLongPress={[Function]}
                onPress={[Function]}
              >
                <Styled(Icon)
                  name="sort-descending"
                  type="MaterialCommunityIcons"
                />
              </Styled(Button)>,
              <Styled(Button)
                onPress={[Function]}
              >
                <Styled(Icon)
                  name="view-list"
                  type="MaterialIcons"
                />
              </Styled(Button)>,
            ],
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Icon)
                  name="menu"
                  type="MaterialIcons"
                />,
                "onPress": [Function],
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "name": "menu",
                  "type": "MaterialIcons",
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Icon)
                  name="refresh"
                  type="MaterialIcons"
                />,
                "onPress": [Function],
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "name": "refresh",
                  "type": "MaterialIcons",
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Icon)
                  name="add"
                  type="MaterialIcons"
                />,
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "name": "add",
                  "type": "MaterialIcons",
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Icon)
                  name="sort-descending"
                  type="MaterialCommunityIcons"
                />,
                "onLayout": [Function],
                "onLongPress": [Function],
                "onPress": [Function],
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "name": "sort-descending",
                  "type": "MaterialCommunityIcons",
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Icon)
                  name="view-list"
                  type="MaterialIcons"
                />,
                "onPress": [Function],
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "name": "view-list",
                  "type": "MaterialIcons",
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing PreviewFooter Component renders in muted 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PreviewFooter
    muted={true}
    onRefresh={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <FilterPopup
          onRef={[Function]}
          position={
            Object {
              "x": 0,
              "y": 0,
            }
          }
        />,
        <Styled(FooterTab)>
          <Styled(Button)
            onPress={[Function]}
          >
            <Styled(Icon)
              name="menu"
              type="MaterialIcons"
            />
          </Styled(Button)>
          <Styled(Button)
            onPress={[Function]}
          >
            <Styled(Icon)
              name="view-list"
              type="MaterialIcons"
            />
          </Styled(Button)>
        </Styled(FooterTab)>,
      ],
      "onLayout": [Function],
      "style": Object {
        "zIndex": 1,
      },
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onRef": [Function],
          "parentHeight": undefined,
          "position": Object {
            "x": 0,
            "y": 0,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Button)
              onPress={[Function]}
            >
              <Styled(Icon)
                name="menu"
                type="MaterialIcons"
              />
            </Styled(Button)>,
            false,
            false,
            false,
            <Styled(Button)
              onPress={[Function]}
            >
              <Styled(Icon)
                name="view-list"
                type="MaterialIcons"
              />
            </Styled(Button)>,
          ],
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Icon)
                name="menu"
                type="MaterialIcons"
              />,
              "onPress": [Function],
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "menu",
                "type": "MaterialIcons",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          false,
          false,
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Icon)
                name="view-list"
                type="MaterialIcons"
              />,
              "onPress": [Function],
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "view-list",
                "type": "MaterialIcons",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <FilterPopup
            onRef={[Function]}
            position={
              Object {
                "x": 0,
                "y": 0,
              }
            }
          />,
          <Styled(FooterTab)>
            <Styled(Button)
              onPress={[Function]}
            >
              <Styled(Icon)
                name="menu"
                type="MaterialIcons"
              />
            </Styled(Button)>
            <Styled(Button)
              onPress={[Function]}
            >
              <Styled(Icon)
                name="view-list"
                type="MaterialIcons"
              />
            </Styled(Button)>
          </Styled(FooterTab)>,
        ],
        "onLayout": [Function],
        "style": Object {
          "zIndex": 1,
        },
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onRef": [Function],
            "parentHeight": undefined,
            "position": Object {
              "x": 0,
              "y": 0,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(Button)
                onPress={[Function]}
              >
                <Styled(Icon)
                  name="menu"
                  type="MaterialIcons"
                />
              </Styled(Button)>,
              false,
              false,
              false,
              <Styled(Button)
                onPress={[Function]}
              >
                <Styled(Icon)
                  name="view-list"
                  type="MaterialIcons"
                />
              </Styled(Button)>,
            ],
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Icon)
                  name="menu"
                  type="MaterialIcons"
                />,
                "onPress": [Function],
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "name": "menu",
                  "type": "MaterialIcons",
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            false,
            false,
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Icon)
                  name="view-list"
                  type="MaterialIcons"
                />,
                "onPress": [Function],
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "name": "view-list",
                  "type": "MaterialIcons",
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
